### PR TITLE
musl exp port

### PR DESCRIPTION
**Pull request by musm**
musl exp port.

### CODE REVIEW

**Comment by musm** [comment]
``` julia
***** Benchmark exp *****

BenchmarkTools.TrialRatio:
  time:             4.165963202610034
  gctime:           9.484611216144136
  memory:           7.298063237745977
  allocs:           1.1357643333333333e6
  time tolerance:   5.00%
  memory tolerance: 1.00%

Details

BenchmarkTools.Trial:
  samples:          48
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  60.25 mb
  allocs estimate:  3407293
  minimum time:     86.70 ms (3.72% GC)
  median time:      94.41 ms (6.50% GC)
  mean time:        105.76 ms (9.50% GC)
  maximum time:     216.52 ms (5.96% GC)

BenchmarkTools.Trial:
  samples:          198
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  8.26 mb
  allocs estimate:  3
  minimum time:     20.87 ms (0.00% GC)
  median time:      22.67 ms (0.00% GC)
  mean time:        25.39 ms (4.17% GC)
  maximum time:     67.43 ms (11.50% GC)

```

**Comment by ararslan** [comment]
Looks like you closed #6. Please update existing PRs rather than opening new ones for the same functionality. This also appears to include your commit for #3. It's best to keep unrelated PRs separate. You can rebase out that commit on this branch.

**Comment by simonbyrne** [comment]
This needs to be rebased against master.

**Comment by JeffBezanson** [comment]
The performance gap is worrying; any sense of what's going on?

**Comment by musm** [comment]
tbh I stopped with this  after I started the sleef port (sleef code is a lot cleaner and the exp fun there is a bit faster but I believe It can be made faster to close the perf gap. I just need some more time to clean things up and audit the code before I get to that).

File: src/utils.jl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        n += 1022
+        if n < -1022
+            x *= 0x1p-1022
+            n += 1022
+            if n < -1022
+                n = -1022
+            end
+        end
+    end
+    u = reinterpret(Float64, (0x3ff+n) % UInt64 << 52)
```
Comment (musm) [inline src/utils.jl:30]: I need to fix the type mismatch here 0x3ff + n  which throws off type inference

**Comment by musm** [comment]
I'm closing this in favor of  the implementation in https://github.com/JuliaMath/Libm.jl/pull/11

**Pull request closed without merging**